### PR TITLE
Upgrade terraform sdk to v1.7.0

### PR DESCRIPTION
**Pull request by davidji99**
Upgrading the SDK from `v1.0.0` to `v1.7.0`. While there were a lot of changes in the SDK since `v1.0`, I didn't see any deprecation warnings in the code base.

@mars review?

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,8 +6,10 @@ require (
...
 	github.com/hashicorp/go-uuid v1.0.1
 	github.com/hashicorp/hcl v0.0.0-20180404174102-ef8a98b0bbce // indirect
-	github.com/hashicorp/terraform-plugin-sdk v1.0.0
+	github.com/hashicorp/terraform-plugin-sdk v1.7.0
 	github.com/heroku/heroku-go/v5 v5.1.0
 	github.com/mitchellh/go-homedir v1.1.0
 	github.com/verybluebot/tarinator-go v0.0.0-20190613183509-5ab4e1193986
 )
+
+go 1.13
```
Comment (davidji99) [inline go.mod]: This was auto added by `go mod` command. Seeing how Go `v1.14` is nearing completion, I don't see any issues in locking to `v1.13`.

**Comment by mars** [comment]
Awesome @davidji99  😄🙌 Looks great. I agree that it's fine to require Go 1.13.

I kicked off a full Hashicorp CI run ⏳

**Comment by davidji99** [comment]
Yep, go ahead @mars.

**Comment by mars** [comment]
Released as [v2.2.2](https://github.com/terraform-providers/terraform-provider-heroku/releases/tag/v2.2.2) 🏁

**Pull request merged**